dll/win32/shell32/CDefViewUtil.cpp

```diff
@@ -37,70 +37,78 @@ class CShellFolderViewCBWrapper :
     public IShellFolderViewCB,
     public CObjectWithSiteBase
 {
-    protected:
-        HWND                    m_hWndMain;
-        CComPtr<IShellFolder>   m_psf;
-        CComPtr<IShellView>     m_psvOuter;
-        LPFNVIEWCALLBACK        m_Callback;
-        FOLDERVIEWMODE          m_FVM;
-        LONG                    m_Events;
-    public:
-        CShellFolderViewCBWrapper() : m_hWndMain(NULL) {}
-
-        HRESULT WINAPI Initialize(LPCSFV psvcbi)
+protected:
+    HWND                    m_hWndMain;
+    PIDLIST_ABSOLUTE        m_Pidl;
+    CComPtr<IShellFolder>   m_psf;
+    CComPtr<IShellView>     m_psvOuter;
+    LPFNVIEWCALLBACK        m_Callback;
+    FOLDERVIEWMODE          m_FVM;
+    LONG                    m_Events;
+
+public:
+    CShellFolderViewCBWrapper() : m_hWndMain(NULL), m_Pidl(NULL) {}
+
+    virtual ~CShellFolderViewCBWrapper()
+    {
+        ILFree(m_Pidl);
+    }
+
+    HRESULT WINAPI Initialize(LPCSFV psvcbi)
+    {
+        m_psf = psvcbi->pshf;
+        m_psvOuter = psvcbi->psvOuter;
+        m_Pidl = psvcbi->pidl ? ILClone(psvcbi->pidl) : NULL;
```

Comment:
Just to be sure: the pidl has to be cloned because it can go away behind ourselves during init time and when it's needed (when `MessageSFVCB` is called)?

Reply:
Correct.

The documentation for SFVM_GETNOTIFY makes it clear we must own this pidl for the lifetime of the callback function. This documentation must have been written by someone with source code access but not full understanding of the code since they suggest which variable name to use which is highly unusual. So now we know that DefView probably calls this `pidlMonitor` 😸